app/scripts/lib/createMetaRPCHandler.js

```diff
@@ -20,6 +60,19 @@ const createMetaRPCHandler = (api, outStream) => {
     let error;
     try {
       result = await api[data.method](...data.params);
+
+      if (
+        isDirty &&
+        store &&
+        !['trackMetaMetricsEvent', 'getState'].includes(data.method)
```

Comment:
true. I'm just not sure that subscribing to the store and setting `isDirty` is actually helping? Since it seems like the `controller.store` is emitting `update` events even when methods like `trackMetaMetricsEvent` or `getState` are called?

Reply:
`trackMetaMetricsEvent` does changes state of metrics controller. But yep looks like `isDirty` will almost always be true.

Reply:
removed the `store.subscribe` and `isDirty` part + removed `trackMetaMetricsEvent` from the array to skip on: https://github.com/MetaMask/metamask-extension/pull/15978/commits/15dc4b6287d4b9b4fecc08535262d42627bf25e6